### ISSUE TITLE
Can not use zip_extract inside thread

### ISSUE THREAD

**Issue by ac2pic** [kuba--/zip#221]
I am trying to use `zip_extract` inside a PS4 application. When I test it on the main thread, there are no errors. However, when I test it inside a detached thread, it crashes the entire application. I am using C++11 std::thread.  I am currently not running it in multiple threads.

**Comment by kuba--** [comment]
Can you share any parts of your code? If the thread is independent (you alloc and dealloc on the same thread), then it's hard to say where and why it happens. 
Any stacktrace, dumped core, ...?

**Comment by ac2pic** [comment]
Here is the line 
https://github.com/Team-Alua/ps4-save-signer/blob/2deb0a9e8f7723f9b776f39d5c3ce73c66412192/ps4-save-signer/cmd_saveextract.cpp#L91

Here is around where I found it to crash:
 https://github.com/kuba--/zip/blob/c30cb27c740a93c681bde8765103c0a0167ecb67/src/zip.c#L283-L292
 
#
# A user thread receives a fatal signal
#
# signal: 11 (SIGSEGV)
# thread ID: 101881
# thread name: 
# proc ID: 108
# proc name: eboot.bin
# reason: page fault (user write data, page not present)
# fault address: 00000007eedab150
#
# registers:
# rax: 0000000000000000  rbx: 0000000000000000
# rcx: 00000007eedbbb6c  rdx: 0000000000400c90
# rsi: 00000007eedbbb70  rdi: 00000007eedbb600
# rbp: 00000007eedbbff0  rsp: 00000007eedab110
# r8 : 00000007eedab5b0  r9 : 00000007eedb35b0
# r10: 0000000000000000  r11: 0000000000000206
# r12: 0000000000000000  r13: 0000000000000000
# r14: 0000000000000000  r15: 0000000000000000
# rip: 0000000000520114  eflags: 00010246
# BrF: 0000000000520b11  BrT: 00000000005200d0
#
# backtrace:
copyin:  has nonsleeping lock
# 0000000000000000
#
# dynamic libraries:
# /app0/eboot.bin
#  text: 0000000000400000:00000000005cc000 r-x
#  data: 00000000005dc000:00000000005e0000 rw-

**Comment by ac2pic** [comment]
Found a potential fix: 
https://github.com/kuba--/zip/blob/c30cb27c740a93c681bde8765103c0a0167ecb67/src/zip.h#L48
Lowering this value to 256 does not cause a crash. So the stack size was the issue.

**Comment by kuba--** [comment]
I understand the stack size problem, but it can be related to any data structure in your application. In this case it can be _path buffer_ , but if you change ordering of allocations it can be something else. Moreover, I don't see any direct connections with threads (do you have set up stack size limits for threads?).
Anyway, I totally agree that 32K is too big in most cases, but on the other hand it's hard to define some upper limits.
We can also rely on `PATH_MAX` from `limits.h`, which on some platforms is `1024` but on others can be `256`.

Long story short, I'm totally fine to change `MAX_PATH` to `1024` if it works for you, but frankly it's just a pre-proc. macro, so you can redefine it in your app (before you include zip.h), or just compile your project with `-DMAX_PATH=256`.

**Comment by ac2pic** [comment]
I did not set up a stack size limit. Every relevant case that it crashes, there is an associated line where it reserves well over 1K on the stack. [Here](https://github.com/kuba--/zip/blob/c30cb27c740a93c681bde8765103c0a0167ecb67/src/miniz.h#L1136) is another example.

**Comment by kuba--** [comment]
@ac2pic I merged PR, where I reduced MAX_PATH to 1K and BUF_SIZE to 8K.
I hope it works for you, otherwise I suggest to redefine MAX_PATH in your build system.

**Comment by ac2pic** [comment]
Okay. Thank you.
